src/sage/matrix/args.pyx

```diff
@@ -1219,12 +1245,14 @@ cdef class MatrixArgs:
         # hurt to do these first.
         if self.entries is None:
             return MA_ENTRIES_ZERO
+        if isinstance(self.entries, (int, float, complex, Integer)):
```

Comment:
what's special about these 4 types?

Reply:
This part was only modified by adding the integer.Integer type, which looks basic enough to be put beside the three others (typical sage code is probably using Integer more often than int, I think?). So for the original motivation for this piece of code, I can only guess the author's aim. Comments help: just above this block is written:
```
        # Check basic Python types. This is very fast, so it doesn't
        # hurt to do these first.
```
So I suppose that the idea was that in any case, these checks "take no time", so why not doing them first even though some of them may be redundant with the rest of the `get_type` process. This can speed up the matrix args process a bit (in fact, the difference is not that small, but since in most reasonable cases `get_type` is only a minor part of the time of an actual matrix creation...):
```
sage: space = MatrixSpace(GF(2), 50, 50)
sage: izero = int(0)
sage: fzero = GF(2)(0)
sage: Izero = 0
sage: %timeit ma = MatrixArgs_init(space, fzero)
885 ns ± 8.95 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
sage: %timeit ma = MatrixArgs_init(space, izero)
585 ns ± 3.31 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
sage: %timeit ma = MatrixArgs_init(space, Izero)
597 ns ± 5.6 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
```